Special abilities PR 5, Split Speed values (#8)
GameBrains/er-core: modules/main-rules/pages/transport.adoc

> = Transportation

Reviewer: This would've always been a better title I think, and it helps to disambiguate from the ability (and from the flyer mission, though we need to improve those titles and filenames later anyway).

I should change the filename to match too - but I will also need to find and update all inbound links to use the new filename, so I'll see what you think of the edits first and sort that out once we're settled.

Author: Looks good to me.

Reviewer: Done, so I'll resolve this. For future reference though, the pull request basically sees the renamed file as 'new' in comparison with the master branch it is comparing to. That's logical of course, but it makes it harder to see what is truly different from master, and it messes with the conversations and suggestions...

We do need to be free to rename files to match titles, but it seems that we also need to be aware of these limitations so we don't get into a pickle with any complex conversations or suggestion that might be in progress. As it is in the case there were only a couple of points to round off.

On some unit data sheets you'll see that the unit has one of two special abilities that relate to transportation: {transport} or {hitch}.
We explain these and other special abilities in more detail elsewhere; but transport is a vital aspect of the Movement phase so we'll discuss it here too.

When we speak of these transportation abilities, we use the terms 'embark/embarked' and 'disembark/disembarked' to describe actions and states that relate to the units that you transport.

Both the {transport} and {hitch} abilities indicate units that can carry other units across the battlefield, but they work in slightly different ways.

== Transportation by units that have the {transport} ability

Units that have the {transport} ability can carry other units across the battlefield.
They can only transport Infantry units -- except where the rules or army lists say otherwise.

To embark or disembark a unit:

* You must sacrifice 5 cm from the movement of the {transport} unit.
* You must sacrifice 5 cm from the movement of the unit that you embark or disembark.

> * You may embark and disembark units in the same turn, the same phase or even in the same move -- so long as all of the involved units have enough movement allowance.

Author: What's a 'movement allowance'?

Reviewer: Pretty much a cop-out until we've finished honing the whole 'move' thing. Potentially 'movement' is more than a single 'move', so your 'movement allowance' is the total of all the moves you're doing right now, if it is more than one move (ie, March orders).

Feel free to suggest an alternative that works.

Author: > Feel free to suggest an alternative that works.

'Movement'? 😜

More seriously, it's what the rest of the page is using for that concept, which is why 'movement allowance' stood out so much.

Author: But it's a very minor issue.

Reviewer: Yeah, I might've had some nuance in mind that set me on a tangent, but I can't think of it now. Maybe a brain fart. Movement will do. :-)
```suggestion
* You may embark and disembark units in the same turn, the same phase or even in the same move -- so long as all of the involved units have enough movement.
```

You may perform no other actions with units that are embarked -- so, you cannot shoot with them, for example.
Obviously, once you disembark them then the normal rules apply again.

[NOTE]
.Placement of disembarked units
====
When you disembark units, measure their movement from the base of the {transport} unit.
If you do not have enough movement to clear the {transport} unit's base then place the disembarked unit in base contact with the {transport} unit.
====

[NOTE.e40k]
.Placement of disembarked units
====
We added this rule to minimise the effects that result from the assortment of different base sizes that players use.
====

== Destruction of {transport} units

If a {transport} unit is destroyed then, just before you remove the unit as a casualty, roll a D6 for each embarked unit:

* *1–3*: The embarked unit suffers a hit.

> * *4–6*: The embarked unit survives -- place it in contact with the destroyed {transport} unit.

Author: This is in reference to the two results of the die roll.

At some point (probably not now), this needs to be restructured so that placement of survivors is not linked to a specific result, but applies to anyone who survives. This will stop units with Save or multiple Damage Capacity from needing extra text.

That's partly why I would have liked to swap it from '4+ to live' to '4+ to get hit', as it allows for a much more concise structure like this:

... roll a D6 for each embarked unit. On a roll of 4+, the unit suffers a hit. If the unit survives, place it in contact with the destroyed {transport} unit.

Reviewer: I see what you mean. Plus, in general I find it much easier to write such rules in a consistent way if we phrase such that the opponent rolls for this type of 'bad stuff' against you. Then the normal '4+/high numbers are good' expectation is met because it is 'good' from your opponent's point of view, just as it is when they roll to hit with their own shooting. Which you've probably thought of too. After all, it doesn't change the actual mechanic, it's still a 50:50 roll.

So, I agree it'd be nice not to hold up this PR for it, but I'm also supportive of the idea that we could try a rewrite of this bit later to 'invert' the result if it will make other elements more tidy and consistent.

Reviewer: Of course, an option that would slot straight into the existing text:

```suggestion
On a roll of 1 to 3, the unit suffers a hit.
If the unit survives, place it in contact with the destroyed {transport} unit.
```

Then remove the {transport} unit as normal.

== Transportation by units that have the {hitch} ability

Units that have the {hitch} ability can carry Infantry units across the battlefield -- they 'hitch a lift'.
That is, they simply grab onto the outside of the vehicle and hang on for a crazy ride over the battlefield.

To use the {hitch} ability to transport an Infantry unit:

* You must embark *and* disembark your Infantry unit in the same phase.

> Embark the unit at any point during your move of the {hitch} unit, and disembark either at the end that same move -- or at the end of another move in the same phase, if the {hitch} unit is on March orders.

Author: This is a rules change, there are no restrictions on when the unit can disembark.

Maybe:

```suggestion
Embark the unit at any point during your move of the {hitch} unit, and disembark at another point during same move -- or during another move in the same phase, if the {hitch} unit is on March orders.
```

Reviewer: Funny thing - I **swear** I saw something about 'at the end of the move' when I was revising the text, and that's why I worded it that particular way. But looking at the E40k source I don't see where it could've crept in from, if indeed it was there (I'll look at the history tomorrow just to satisfy my bafflement). I'm off to bed shortly, but good catch, I'll check your suggestion and fix tomorrow.

Reviewer: Aha! It seems that I'm not entirely mad! :-D Previously, as from commit [e9e402b](https://github.com/GameBrains/er-core/pull/8/commits/e9e402becb9048a6b0c17e513b5f8ba133d158bb), it said:

`* Any embarked unit *must* disembark at the end of the {hitch} unit's move, or at the end of its last move during a March order.`

And... that was actually your commit, you bugger! ;-D

Prior to that it said:

`* You can simply embark and disembark units as you move the vehicle -- there is no cost to the {transport} unit's movement.`

Ah well, stuff happens. I'll look at your latest suggestion next.

Author: > And... that was actually your commit, you bugger! ;-D

Doh! Oops. :-(

Units that have the {transport} special ability can carry other units across the battlefield.

* To embark or disembark other units (or cargo) costs the {transport} unit 5cm of its movement.
* An embarking or disembarking unit must also pay 5cm of its movement.
* You may embark and disembark units in the same turn, the same phase or even in the same move -- so long as all of the involved units have enough movement.
* You may perform no other actions with units that are embarked -- so, you cannot shoot with them, for example.

> Obviously, once you disembark them then the normal rules apply again.

Author: I just remembered some discussions about how to measure movement when disembarking, and whether you should place the unit then measure, or measure and then place the unit.

The main note is a bit wordy, let me know if you can think of something better.

```suggestion
Obviously, once you disembark them then the normal rules apply again.

NOTE: Measure the movement of disembarking units from the base of the {transport} unit. If there is not enough movement to clear the {transport} unit's base, place the disembarking unit in base contact with the {transport} unit.

[NOTE.e40k]
====
This was not specified in _Epic 40,000_, but reduces the effects of different size bases.
====
```

Reviewer: I don't 100% understand what you're trying to say here, so may be good to talk it through. I'm sure we've discussed it before, but I've forgotten the crux of it.  I intend to make some other edits here anyway, so I'll push those later and then we can hone it some more to get across the exact meaning you intend in the best way.

Author: The crux of it is that if you place a disembarking unit in base contact with the Transport, and **then** measure the disembarking unit's movement, different base sizes have a substantial effect on how far the unit can move. 

For example a unit on a 20mm square base unit will lose a couple of centimetres of movement compared to a unit on a 40x12.5mm strip base.

Reviewer: I'll resolve this conversation as I've used this suggestion as the core for changes which now appear below.

Reviewer: >The crux of it is ...

Lol, I think you replied before I posted my message to say I would mark resolved because the new content is below. I'll unresolve so I have a prompt to look at it again. 

Gotcha, I remember now. I'll wordsmith a little more to make this more explicit.

Author: That looks good, resolving the conversation!

Reviewer: Yeah, I didn't get to further word-smithing yet to make that point more explicit. But now that you've reminded me of the crux of the point and I look at the word-smithing I already did, I agree that it is probably adequate and we can get away with not making it more explicit.

Though frankly I don't know why anyone would ever place the unit and then measure. To me, that is obviously wrong -- as wrong as measuring from the front of the base to the back where again you'd gain the width of the base.

Something further occurs to me. There appears to be no reason why you couldn't clear the transport unit's base - you must hold back enough movement to do it, on top of the 5cm cost for embarking/disembarking. We could make *that* explicit. But where the problem really lies IIRC is when you have *lots* of units such that potentially you can't fit them all around the transport (Orkeosaurus). In that case, it isn't clearing the base of the transport that is the problem, it's running out of space even when you've tried to do it all right, isn't it?

Author: P.S. As far as the actual pull request is concerned, I think this can be marked as resolved.

Reviewer: Some interesting points, and definitely room for some more nuance here that we might want to explain eventually. But I agree, let's resolve this for now and revisit in future if and when we feel the urge.

Units that have the {transport} special ability can carry other units across the battlefield.

* To embark or disembark other units (or cargo) costs the {transport} unit 5cm of its movement.
* An embarking or disembarking unit must also pay 5cm of its movement.
* You may embark and disembark units in the same turn, the same phase or even in the same move -- so long as all of the involved units have enough movement.
* You may perform no other actions with units that are embarked -- so, you cannot shoot with them, for example.
Obviously, once you disembark them then the normal rules apply again.

[NOTE]
.Placement of disembarked units
====
When you disembark units, measure their movement from the base of the {transport} unit.
If you do not have enough movement to clear the {transport} unit's base then place the disembarked unit in base contact with the {transport} unit.

> ====

Reviewer: I'm editing this topic at the moment and will share my efforts soon. But on these particular lines, I wonder as follows:

As written, might this permit cases where players could have sacrificed some movement, but don't because the rules say they can put the units in base contact anyway - so they use that to get a 'free' gain?

In general, I feel like perhaps there should be some concept of not exceeding the 'calculated maximum move you could get from the transport plus the embarked unit in combination' -- even if this means you must move transport unit a bit less or something to compensate rather than adding on the width of the disembarked unit's base and thus going beyond the calculated maximum move.

If there is something in this, I'd be happy for us to create an issue to tackle this properly later - just wanted to raise the point and get your thoughts.

Author: > As written, might this permit cases where players could have sacrificed some movement, but don't because the rules say they can put the units in base contact anyway - so they use that to get a 'free' gain?

As far as I can see, this would require them to have a base longer than 5 cm. 

Don't forget that the unit is **already** sacrificing 5 cm of move just to be able to disembark. This is another reason why I automatically assumed that you 'deploy then measure' - as far as I was concerned, the unit had paid a movement penalty to get out, why would the unit have to pay a **second** movement penalty by measuring from the transport to 'move' out?

As a reminder, don't forget that what we currently have is already going to be regarded as a controversial rules change by some players, and if they've played 'deploy then measure' for a long time it's going to be unpopular, at best.

> In general, I feel like perhaps there should be some concept of not exceeding the 'calculated maximum move you could get from the transport plus the embarked unit in combination' -- even if this means you must move transport unit a bit less or something to compensate rather than adding on the width of the disembarked unit's base and thus going beyond the calculated maximum move.

My gut reaction is that this part of the map is marked 'Here be Dragons'. It's likely to be hard to write clear concise rules for, the overall subject is already controversial/generates lots of arguments, it's unlikely to get widespread support from the players (and that includes me), and it adds even more inconsistencies with Hitch. Or you apply it to Hitch as well, and screw over the Orks even more.

Paraphrasing...

'You know that unstoppable Battlefortress war engine that isn't slowed down at all when Da Boyz leap suicidally off the side?'
'Yer.'
'Well, it just slowed down by 4 cm because some of Da Boyz want to leap suicidally off the front instead of the side.'

P.S. In places where I'd normally use italics for emphasis I'm using bold on Github for consistency with the style guide.

Reviewer: >Don't forget that the unit is already sacrificing 5 cm of move just to be able to disembark.

I always rationalised the 5cm cost as a cost in **time** to embark/disembark, which seems reasonable and logical to me - not a 'compensation' for the width of the base, as you appear to see it. Though I can see that perhaps that makes sense from the point of view of game mechanics, but then why pay it twice?

But yeah, okay, let's skip this. The very fact that this part of the rules is controversial and creates arguments is exactly why I'm thinking that perhaps we could at least suggest 'ways to play this to make it more consistent or fair' -- but that could be in another section in future perhaps, especially as it might be hard to make concise, as you say.

It really irritates me when people 'conveniently' gain extra move for their units by virtue of the width of their base - so personally I tend to play it so that no individual unit can move farther in total than the maximum you can calculate that they should move. That seems the only fair way to me, and if that means moving your Battlefortress back a bit then so be it. If the Orks are massively disadvantaged in some way, then fix that rather than focus on loss of a gain that wasn't earned anyway.

But yes, different systems, different views. I've taken a view that seems the most consistent and fair to me.

I'll resolve this as 'nothing to do here', but if we do ever do a section for 'suggestions you can pick and choose for things that might otherwise be contentious or that don't render down to nice concise rules' then I think this is a contender for more attention!
